### ISSUE TITLE
Update release process to be automatable

### ISSUE THREAD

**Issue by SteveALee** [mulberrysymbols/mulberry-symbols#35]
Currently the release process is manual and a bit messy

Currently the site and zip are entwined and especially coupled by the category files. This is not really clearly expressed in the release documentation.

Thus we need to decide on the candence for each and if they need to be separate or are always released together. For example

* Zip and category files should be tightly version controlled and formally released together 
* Rest of website could be loosly managed - ie no versioning and just updated by push to master

**Comment by gavinhenderson** [comment]
Personally I would prefer the whole thing to be automated, and shouldn't be too hard. Its ashame there is not Github Actions support for orgs yet as it would do the job great.

The build scripts are quite awkwardly tied right now. For example you cant create the pdf or html independently right now. If its alright I might do a bit of refactor of the scripts which should make it easier to automate.

**Comment by SteveALee** [comment]
+1 to refactoring. I was thinking that recently. AiM for loose coupling.

I must try the actions. they would be ideal. Shame about orgs though.

Thanks.

**Comment by gavinhenderson** [comment]
I am quite busy for the next week or two so it might take me a while to get round to it, but I will

Actually turns out you can get [actions for orgs!](https://github.com/features/actions/signup/). Just have to join the queue to get into the beta. Even if we never use it would be nice to have it as a options.

**Comment by SteveALee** [comment]
Sure -it's low priority. Even though should have been done at the start (I cut corners as originally it was simple)

I have a beta but not tried it yet (on my list) - will see if can activate it for the org so you can test too.

**Comment by SteveALee** [comment]
@gavinhenderson As GitHub recently opened up including actions and workflows this might be fully do-able now with just GitHub? Is this something you fancy looking at?

**Comment by ha6000** [comment]
The reason I created the converter was for Microsoft word

**Comment by ha6000** [comment]
I have some prior knowledge of github actions. I can maybe help out here. We could use your generator scripts and just upload them with this action https://github.com/marketplace/actions/upload-a-release-asset

**Comment by ha6000** [comment]
Got the bare bones out of the way, just need to actually upload it
https://github.com/ha6000/mulberry-symbols/blob/master/.github/workflows/build.yml

**Comment by ha6000** [comment]
I successfully were able to upload the zip, now just the rest,

**Comment by ha6000** [comment]
I am able to upload all the files that is in the root directory, but everything in categories/* I am unable to  add. From my research the module I am using, uses this https://www.npmjs.com/package/glob for globs. I can upload the categories in a second step.

**Comment by ha6000** [comment]
If you ignore the fact that it uploads categories separate, it works.

**Comment by SteveALee** [comment]
@ha6000 thanks for PR #56. Great work!

I'll test it out ASAP but am not sure how to actually do that without running it against a live repro. I guess I can clone and run as the workflow doesn't ref the repo explicitally so should work in any.

Does it matter that the second set of files are added separately? Don't we still end up with a single release with all the correct files?

I think the version may be encoded in a file some place as well as the release tag. I'll check.

I'll check out the dep `svenstaro/upload-release-action@1.1.0` to just for good practice :)

**Comment by ha6000** [comment]
Yes, I am pretty sure we end up with same results, just everything does not get uploaded at the same time.

**Comment by ha6000** [comment]
I also said I was going to use https://github.com/marketplace/actions/upload-a-release-asset but they needed some upload url. And I am not sure how to get it from the github.ref so I decided to use that other one
